### PR TITLE
Fix windows pipe permissions.

### PR DESCRIPTION
**Pull request by rogerta**
When the agent is not running in user specific mode, it should
allow any process running in the interactive user's logon session
to send content analysis requests.  Otherwise the agent should
allow any process that is running as a logged on user to send
requests.

### CODE REVIEW

File: common/utils_win.cc [hunk 1]
```diff
@@ -59,11 +59,43 @@ std::string GetPipeName(const std::string& base, bool user_specific) {
...
+  //
+  // If `user_specific` is true, a different agent instance is used for each
+  // OS user, so only allow the interactive logged on user to reads and write
+  // messages to the pipe.  Otherwise only one agent instance used used for all
+  // OS users and all authenticated logged on users can reads and write
+  // messages.
+  //
+  // See https://docs.microsoft.com/en-us/windows/win32/secauthz/security-descriptor-definition-language
+  // for a description of this string format.
+  const char* kDaclEveryone = "D:"
```
Comment (dominiquefc) [inline common/utils_win.cc:80]: nit: const => constexpr

File: common/utils_win.cc [hunk 2]
```diff
@@ -74,11 +106,13 @@ DWORD CreatePipe(
   *handle = CreateNamedPipeA(name.c_str(), mode,
     PIPE_TYPE_MESSAGE | PIPE_READMODE_MESSAGE | PIPE_WAIT |
     PIPE_REJECT_REMOTE_CLIENTS, PIPE_UNLIMITED_INSTANCES, kBufferSize,
-    kBufferSize, 0, /*securityAttr=*/nullptr);
+    kBufferSize, 0, &sa);
   if (*handle == INVALID_HANDLE_VALUE) {
     err = GetLastError();
   }
 
+  LocalFree(sa.lpSecurityDescriptor);
```
Comment (dominiquefc) [inline common/utils_win.cc:114]: Freeing this here implies that the created pipe no longer needs access to `sa`.
Could you add a comment detailing that, or correct me if I'm reading this wrong?

Comment (rogerta) [reply common/utils_win.cc:116]: Correct no longer needed.  Added comment.

File: common/utils_win.cc (hunk 1 above)
Comment (rogerta) [reply common/utils_win.cc]: Done.

File: common/utils_win.cc (hunk 2 above)
Comment (rogerta) [reply common/utils_win.cc:116]: Correct no longer needed.  Added comment.

**Review by dominiquefc — APPROVED** [review]

**Pull request merged**